src/Storages/MergeTree/IDataPartStorage.h

```diff
@@ -129,7 +129,11 @@ class IDataPartStorage
 
     /// Get a name like 'prefix_partdir_tryN' which does not exist in a root dir.
     /// TODO: remove it.
-    virtual std::string getRelativePathForPrefix(Poco::Logger * log, const String & prefix, bool detached) const = 0;
+    virtual std::optional<String> getRelativePathForPrefix(Poco::Logger * log, const String & prefix, bool detached, bool broken) const = 0;
```

Comment:
Please, add a comment what should be the semantic for `std::nullopt`. E.g. what we gonna do with it :)

Reply:
Also not clear what combination of detached/broken is possible.
Maybe replace two this flags to Enum?

Reply:
But we also have `getRelativePathForDetachedPart` and it has only one flag (`broken`). We can use enum for `getRelativePathForPrefix`, but for `getRelativePathForDetachedPart` we have two options:
 - pass enum instead of flag too, but it will not be clear which values of enum are possible
 - keep one flag and convert it to enum, but I don't like mixing enum and flags

So both options look bad